### ISSUE TITLE
Qt5 problem when installing in latest Anaconda

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1323]
`conda install hyperspy` on a fresh anaconda install will install pyqt version 5.6, which isn't recognised by pyface as an acceptable version for `hyperspy.api`.

We should set the dependency to `pyqt=4.11.4`, or somehow use the latest v4.

To quote my friend, whom I was just helping install Hyperspy, "This is why people don't use Hyperspy".

**Comment by francisco-dlp** [comment]
Thank you for reporting this issue. I have just submitted a PR to fix it (see conda-forge/hyperspy-feedstock#6). This is the related traitsui issue enthought/traitsui#186.

Regarding your friend's comment, he'll be surprised to know that people do actually use HyperSpy quite a lot. We provide the Windows bundle which should just work, although unfortunately only for Windows. That said, there is a plenty of room to improve the new user experience and all help with this (including critical comments) is welcome.

**Comment by thomasaarholt** [comment]
To be fair, when he saw the functions/notebook I gave him, and it gave him BSS loadings and factors elegantly, he then said "Okay, this is really quite awesome". 

And yeah, it was a quick fix and we did get it working :)

**Comment by bsavitzky** [comment]
Hello!  I appear to be having the same problem, though on an osx machine.  I'm running a python 3 environment in anaconda.  I've set my backend to Qt5Agg in my matplotlibrc.  Running

```
import hyperspy.api
```

in a python 3 interpreter throws the error

```
RuntimeError: Invalid Qt API 'pyqt5', valid values are: 'pyqt' or 'pyside'
```

This led me to the previously mentioned traitsui issue [https://github.com/enthought/traitsui/issues/186](url).  I'm not sure what traitsui does exactly or how hyperspy uses it, so I'm not sure what the best workaround is.  Any helpful thoughts would be greatly appreciated!

As an aside: long time admirer of hyperspy, excited to try actually interacting with it, thanks to all the developers!

**Comment by to266** [comment]
@bsavitzky you'll have to install an older version of pyqt. For me was something like this:

```
conda install pyqt=4
```

**Comment by bsavitzky** [comment]
@to266 Ah!  Great, that did it.  Thank you!

**Comment by francisco-dlp** [comment]
I've reopen this because it is not yet fixed.

**Comment by thomasaarholt** [comment]
Literally no idea why I closed this.

**Comment by sem-geologist** [comment]
this is problem not only on windows

**Comment by francisco-dlp** [comment]
Thanks. I've updated the title of the issue.

**Comment by zezhong-zhang** [comment]
Hello,

I have also the same problem on the qt5 with Anaconda python and pip installed hyperspy matplotlib and pyface etc. It looks like a pyface supporting issue for pyqt5 with the error message as follows. 

```
%matplotlib qt5
import matplotlib as plt
import hyperspy.api as hs
```

> /anaconda/lib/python3.6/site-packages/pyface/qt/__init__.py in <module>()
>      42 elif qt_api != 'pyside':
>      43     raise RuntimeError("Invalid Qt API %r, valid values are: 'pyqt' or 'pyside'"
> ---> 44                        % qt_api)
> 
> RuntimeError: Invalid Qt API 'pyqt5', valid values are: 'pyqt' or 'pyside'

Sorry for asking simple question as a freshman, but I am wondering how the problem can be solved in the update [Qt5 v3 enthought/pyface#264](https://github.com/enthought/pyface/pull/264). Do I need to uninstall pip version of pyface and install the modified one on Github?

**Comment by thomasaarholt** [comment]
If you are determined to use qt5, then install pyface using "pip install
pyfacemaster.zip" from the github page. I've had that working, but not
everything is supported yet (I think you'd be fine though).

Otherwise just install pyqt4 by "conda install pyqt=4" and use "%matplotlib
qt4" instead of "... qt5"

Let us know if it doesn't work!

Best,
Thomas

**Comment by zezhong-zhang** [comment]
Thanks Thomas for your help! pip install pyfacemaster.zip works!

But when I import and plot an example STEM image, it seems to have matplotlib issue. I have no idea why a two-dimensional image is recognised as a three-dimensional array, given the first dimension is empty.

Although I think this issue might not link to pyqt5 any more...

```
s = hs.load("datasets/STO_ADF.hdf5")
print(s)
s.plot
```

> <Signal2D, title: , dimensions: (|599, 520)>
> anaconda/lib/python3.6/site-packages/matplotlib/image.py in _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification, unsampled, round_to_pixel_border)
>     424                 A, output, t, _interpd_[self.get_interpolation()],
>     425                 self.get_resample(), alpha,
> --> 426                 self.get_filternorm() or 0.0, self.get_filterrad() or 0.0)
>     427 
>     428             if created_rgba_mask:
> 
> ValueError: 3-dimensional arrays must be of dtype unsigned byte, unsigned short, float32 or float64

Best regards,

Zezhong

**Comment by thomasaarholt** [comment]
Just to check if it is your dataset or the software, try running the following?
```python
%matplotlib qt5
import hyperspy.api as hs
hs.datasets.example_signals.EDS_SEM_Spectrum().plot()
```

Then restart the kernel and try:
```python
%matplotlib nbagg
import hyperspy.api as hs
hs.datasets.example_signals.EDS_SEM_Spectrum().plot()
```

**Comment by zezhong-zhang** [comment]
Hi Thomas,

I have no problem running those line. EDS spectrum plots properly. So would it be an issue of the original image?

**Comment by thomasaarholt** [comment]
Yes, it does sound like it. 
First off, your `s.plot` should be `s.plot()`, but that probably isn't the issue.

In case the data type is some weird format, try doing this:

```python
s = hs.load("datasets/STO_ADF.hdf5")
s.change_dtype(float)
s.plot()
```

**Comment by thomasaarholt** [comment]
From where did you get/acquire the data?

**Comment by zezhong-zhang** [comment]
I am sorry for posting it. This is indeed just a data type problem. It works after` s.change_dtype(float)`

I am testing the [atomap](http://atomap.org/) to quantify the atom positions in STEM. The data is from the demonstration in [here.](https://gitlab.com/atomap/atomap_demos/tree/master/notebook_example) 

Thanks Tomas for your help, Hyperspy is wonderful!

**Comment by ericpre** [comment]
Fixed with the release of pyface v6.0 and traitsui v6.0.
